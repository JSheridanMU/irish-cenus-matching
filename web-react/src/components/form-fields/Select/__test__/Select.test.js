import * as React from 'react'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Select from '.././Select'

afterEach(cleanup)
const options = [
  { id: '1901', title: '1901' },
  { id: '1911', title: '1911' },
]

test('renders without crashing', () => {
  render(<Select label="Test Label" options={options} />)
})

test('renders select with label', () => {
  const { getAllByText } = render(
    <Select label="Test Label" options={options} />
  )

  getAllByText('Test Label')
})

test('matches snapshot', () => {
  const tree = renderer
    .create(<Select label="Test Label" options={options} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
