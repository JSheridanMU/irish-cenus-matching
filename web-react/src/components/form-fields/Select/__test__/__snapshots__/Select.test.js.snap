// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
    data-shrink={false}
  >
    Test Label
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-11"
      >
        <span>
          Test Label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
